### DISCUSSION TITLE
Is there a way to determine the evaluation time of a script using the performance API?

### DISCUSSION THREAD

**Discussion by alshdavid** [mdn/mdn-community#537] [Content]
I am working on a project where the compilation output produces hundreds of small dynamic bundles. The performance of these bundles is obtained by placing the following automatically at compile time:

```javascript
performance.mark('bundle-name:eval-start') 

// ...
// code
// ...

performance.mark('bundle-name:eval-end')
```

Looking at `performance.getEntries()`, I can see the resource timings for scripts, however I cannot see the any data on runtime performance, only network information.

Does the performance API offer the ability to inspect any runtime performance metrics?

**Comment by Elchi3** [comment]
Would a `PerformanceObserver` help you? https://developer.mozilla.org/en-US/docs/Web/API/Performance_API/User_timing#observing_performance_measures
